neqo-transport/src/connection.rs

```diff
@@ -511,6 +512,20 @@ impl Connection {
         Ok(())
     }
 
+    /// Enable a set of ciphers.
+    pub fn enable_ciphers(&mut self, ciphers: &[Cipher]) -> Res<()> {
+        if self.role() != Role::Client {
+            qerror!([self], "Cannot enable ciphers in role {:?}", self.role);
+            return Err(Error::ConnectionState);
+        }
+        if self.state != State::Init {
```

Comment:
Can you also add a similar check in SecretAgent::enable_ciphers?  Belt and braces.